RecoPPS/Local/plugins/CTPPSPixelLocalTrackProducer.cc

```diff
@@ -170,30 +170,62 @@ void CTPPSPixelLocalTrackProducer::produce(edm::Event &iEvent, const edm::EventS
+
   if (!recHits->empty()) {
     const auto &mask = iSetup.getData(tokenCTPPSPixelAnalysisMask_);
 
     // Read Mask checking if 45-220-far is masked as bad and needs special treatment
     std::map<uint32_t, CTPPSPixelROCAnalysisMask> const &maschera = mask.analysisMask;
 
-    bool mask_45_220[6][6] = {{false}};
+    // vector of masked rocs
+    bool maskV[4][6][6] = {{{false}}};
+
     for (auto const &det : maschera) {
       CTPPSPixelDetId detId(det.first);
       unsigned int rocNum = (det.first & rocMask) >> rocOffset;
       if (rocNum > 5 || detId.plane() > 5)
         throw cms::Exception("InvalidRocOrPlaneNumber") << "roc number from mask: " << rocNum;
 
-      if (detId.arm() == 0 && detId.station() == 2 && detId.rp() == 3) {  // pot 45-220-far
-        if (det.second.maskedPixels.size() == rocSizeInPixels) {          // roc fully masked
-          mask_45_220[detId.plane()][rocNum] = true;
+      if (detId.arm() == 0 && detId.rp() == 3) {
+        if (detId.station() == 2) {                                                                // pot 45-220-far
+          if (det.second.maskedPixels.size() == rocSizeInPixels || det.second.fullMask == true) {  // roc fully masked
+            maskV[0][detId.plane()][rocNum] = true;
+          }
+        } else if (detId.station() == 0) {                                                         // pot 45-210-far
+          if (det.second.maskedPixels.size() == rocSizeInPixels || det.second.fullMask == true) {  // roc fully masked
+            maskV[1][detId.plane()][rocNum] = true;
+          }
+        }
+      } else if (detId.arm() == 1 && detId.rp() == 3) {
+        if (detId.station() == 0) {                                                                // pot 56-210-far
+          if (det.second.maskedPixels.size() == rocSizeInPixels || det.second.fullMask == true) {  // roc fully masked
+            maskV[2][detId.plane()][rocNum] = true;
+          }
+        } else if (detId.station() == 2) {                                                         // pot 56-220-far
+          if (det.second.maskedPixels.size() == rocSizeInPixels || det.second.fullMask == true) {  // roc fully masked
+            maskV[3][detId.plane()][rocNum] = true;
+          }
         }
       }
     }
+    // search for specific pattern that requires special reconstruction (use of two plane tracks)
 
-    // search for specific pattern that requires special reconstruction (isBadPot)
-    isBadPot_45_220 = mask_45_220[1][4] && mask_45_220[1][5] && mask_45_220[2][4] && mask_45_220[2][5] &&
-                      mask_45_220[3][4] && mask_45_220[3][5] && mask_45_220[4][4] && mask_45_220[4][5];
+    for (unsigned int i = 0; i < 4; i++) {
+      unsigned int numberOfMaskedPlanes = 0;
+      for (unsigned int j = 0; j < 6; j++) {
+        if (maskV[i][j][0] && maskV[i][j][1] && maskV[i][j][4] && maskV[i][j][5])
+          numberOfMaskedPlanes++;
+      }
+      if (numberOfMaskedPlanes == 4)
```

Comment:
What happens if the the number of masked planes is 5 or 6 (which is possible, in principle)? Is it not a "bad pot"?

Reply:
If you have 5 masked plane the pot is not bad but TOO bad, so you don’t activate the special 2-plane reco and go with standard reco that, as it has always been, does not even try to reconstruct a track (with one plane only wouldn’t make sense).

Reply:
Still, don't you think this should be >= 4 instead here? Where is the "too bad pot" flagged?

Reply:
> Still, don't you think this should be >= 4 instead here? Where is the "too bad pot" flagged?

@fabferro rethinking at it, it must remain `== 4`, given the logic in the code.
Please just confirm that if there are more than 4 masked planes a dedicated treatment and/or a protection still exist.
Probably, a more intuitive name for this variable could be `isBadButRecoverablePotV`, but I understand that as such it is rather ugly: any better suggestion, still rendering the idea that those pots are "bad but not too much to throw them away"?

Reply:
@perrotta If the pot has >4 planes dead/masked the track reconstruction algorithm stops at a certain point because it requires at least 3 hits in 3 different planes to fit something. With only one active plane it will likely stop even before, at the level of pattern recognition.
If you believe that "isBadPot" is a misleading name I can change it in "is2PlanePot". 
If you consider it more proper I can prepare the change.